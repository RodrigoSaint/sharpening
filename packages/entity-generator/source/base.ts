export type FieldType = "string" | "int" | "decimal" | "date";

export interface Field {
  name: string;
  isRequired: boolean;
  type: FieldType;
}

export type EntityAction =
  | "pagination"
  | "list"
  | "detail"
  | "create"
  | "update"
  | "delete";

export interface Entity {
  name: string;
  fieldCollection: Field[];
  actionCollection?: EntityAction[];
  formFieldCollection?: string[];
  displayFieldCollection?: string[];
}
