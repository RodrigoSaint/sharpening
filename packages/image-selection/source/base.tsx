import React, { useState } from "react";
import Modal from "@rodrigosaint/modal";
import ManagedImage from "@rodrigosaint/managed-image";

import ImageSelectionModalContent, {
  ImageSelectionModalContentProps,
} from "./content";

export interface ImageSelectionProps<T>
  extends Omit<ImageSelectionModalContentProps<T>, "close"> {
  renderWithoutImage: () => React.ReactNode;
  getImageUrl: (item: T) => string;
  header: string | React.ReactNode;
  initialValue?: T;
}

export default function ImageSelection<T>(props: ImageSelectionProps<T>) {
  const [src, setSrc] = useState<string>(
    props.initialValue ? props.getImageUrl(props.initialValue) : undefined
  );

  const onSelected = React.useCallback(
    (item: T) => {
      setSrc(props.getImageUrl(item));
      props.onSelected(item);
    },
    [props.onSelected, props.getImageUrl]
  );

  return (
    <>
      <Modal
        renderTrigger={() => (
          <ManagedImage
            src={src}
            renderWithoutImage={props.renderWithoutImage}
            renderOnError={props.renderWithoutImage}
          />
        )}
        renderContent={(close) => (
          <ImageSelectionModalContent
            {...props}
            onSelected={onSelected}
            close={close}
          />
        )}
        header={props.header}
      />
    </>
  );
}
