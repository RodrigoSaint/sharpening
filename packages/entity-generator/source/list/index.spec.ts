import { Entity } from "../base";
import { getPagination, getList, getCard } from ".";

const entity: Entity = {
  name: "User",
  fieldCollection: [
    {
      name: "name",
      isRequired: true,
      type: "string",
    },
    {
      name: "email",
      isRequired: false,
      type: "string",
    },
    {
      name: "age",
      isRequired: false,
      type: "int",
    },
  ],
  displayFieldCollection: ["name", "email"],
};

describe("List", () => {
  it("renders pagination properly", () => {
    expect(getPagination(entity)).toMatchSnapshot("getPagination");
  });
  it("renders list properly", () => {
    expect(getList(entity)).toMatchSnapshot("getList");
  });
  it("renders list properly", () => {
    expect(getCard(entity)).toMatchSnapshot("getCard");
  });
});
