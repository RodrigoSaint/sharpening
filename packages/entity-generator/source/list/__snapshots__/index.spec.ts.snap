// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List: getListHook 1`] = `
"import { useCallback } from \\"react\\";
import { useQuery } from \\"@rodrigosaint/graphql\\";

import { User } from \\"entity/user\\";

const USER_LIST_QUERY = \`
  query {
    userCollection {
        name
        email
    }
  }
\`;

type UserList = User[];

export default function useUserList() {
  return (
    useQuery < {}, { userCollection: UserList } > (\\"user\\", USER_LIST_QUERY, {})
  );
}
"
`;

exports[`List Paginate: getPaginationHook 1`] = `
"import { useCallback } from \\"react\\";
import { Page } from \\"@rodrigosaint/pagination\\";
import { queryGraphql } from \\"@rodrigosaint/graphql\\";

import { User } from \\"entity/user\\";

const USER_PAGINATION_QUERY = \`
  query ($page: Int!) {
    userPagination(page: $page) {
      currentPage
      pageCount
      collection {
        name
        email
      }
    }
  }
\`;

export default function useUserPagination() {
  const query = useCallback(
    (page: number) =>
      queryGraphql < Page < User >> (USER_PAGINATION_QUERY, { page }),
    []
  );
  return { query };
}
"
`;

exports[`List renders list properly: getCard 1`] = `
"import Card from \\"@rodrigosaint/card\\";

import { User } from \\"entity/user\\";

interface UserCardProps {
  user: User;
}

export default function UserCard({ user }: UserCardProps) {
  return (
    <Card>
      {user.name}
      {user.email}
    </Card>
  );
}
"
`;

exports[`List renders list properly: getList 1`] = `
"import { User } from \\"entity/user\\";
import List from \\"@rodrigosaint/list\\";

import UserCard from \\"component/user/card\\";

interface UserListProps {
  collection: User[];
}

export default function UserList({ query }: UserListProps) {
  return (
    <List
      collection={collection}
      render={(user) => <UserCard user={user} />}
      getKey={(user) => user._id}
    />
  );
}
"
`;

exports[`List renders pagination properly: getPagination 1`] = `
"import { User } from \\"entity/user\\";
import { ManagedPagination, Page } from \\"@rodrigosaint/pagination\\";

import UserCard from \\"component/user/card\\";

interface UserPaginationProps {
  query: (pageNumber: Number) => Promise<Page<User>>;
}

export default function UserPagination({ query }: UserPaginationProps) {
  return (
    <ManagedPagination
      query={query}
      render={(user) => <UserCard user={user} />}
      getKey={(user) => user._id}
      renderSkeleton={() => <UserCard />}
    />
  );
}
"
`;
